app/controllers/courses_controller.rb

```diff
@@ -12,21 +18,43 @@ def show
   end
 
   def update
+    if !current_user.admin?
```

Comment:
Technically we don't even need this check because the check on the `teacher_id` will handle this for us. But I guess this reduces a db roundtrip, so it's good.